### ISSUE TITLE
Test fail on MacOS because of gnused feature

### ISSUE THREAD

**Issue by mrsdizzie** [wp-cli/wp-cli#5723]
When running behat tests on macOS I saw this failure:

```
--- Failed steps:

001 Scenario: BOM found in wp-config.php file                                 # features/config.feature:548
      And I run `sed -i '1s/^\(\xef\xbb\xbf\)\?/\xef\xbb\xbf/' wp-config.php` # features/config.feature:566
        $ sed -i '1s/^\(\xef\xbb\xbf\)\?/\xef\xbb\xbf/' wp-config.php

        sed: -I or -i may not be used with stdin
        cwd: /var/folders/nc/16hfjyf9517918dd0dz2g7tr0000gn/T/wp-cli-test-run--63bd7a0a590628.51883440/
        run time: 0.017526865005493
        exit status: 1 (RuntimeException)
```
Which comes from https://github.com/wp-cli/wp-cli/blob/f094a7eeec1213d8e0dc35f30949f22c8f4f3db3/features/config.feature#L566

Some scripts try to get around this with a check like: https://github.com/wp-cli/sample-plugin/blob/master/bin/install-wp-tests.sh#L96-L101 though I'm not sure what is supported in these .feature files

**Comment by danielbachhuber** [comment]
`sed -i.bak` is pretty common across the project:

<img width="589" alt="image" src="https://user-images.githubusercontent.com/36432/211682445-b63c8b05-f06e-4b14-8ccc-7680f48affc6.png">

When I apply this change:

```diff
diff --git a/features/config.feature b/features/config.feature
index 55af206b..ce3da1a6 100644
--- a/features/config.feature
+++ b/features/config.feature
@@ -545,6 +545,7 @@ Feature: Have a config file
       https://example.com
       """

+  @daniel
   Scenario: BOM found in wp-config.php file
     Given a WP installation
     And a wp-config.php file:
@@ -563,7 +564,7 @@ Feature: Have a config file
       /** Sets up WordPress vars and included files. */
       require_once(ABSPATH . 'wp-settings.php');
       """
-    And I run `sed -i '1s/^\(\xef\xbb\xbf\)\?/\xef\xbb\xbf/' wp-config.php`
+    And I run `sed -i.bak '1s/^\(\xef\xbb\xbf\)\?/\xef\xbb\xbf/' wp-config.php`

     When I try `wp core is-installed`
     Then STDERR should not contain:
```

It seems `sed -i.bak` runs fine but something about the regex fails:

```
$ composer behat -- --tags @daniel
> run-behat-tests '--tags' '@daniel'
.....F

--- Failed steps:

001 Scenario: BOM found in wp-config.php file # features/config.feature:549
      And STDERR should contain:              # features/config.feature:574
        $ wp core is-installed

        cwd: /var/folders/6t/fzd50mwd57766jb0qdj2vl100000gn/T/wp-cli-test-run--63bdf1b3b9bcd9.55679812/
        run time: 0.24210500717163
        exit status: 0 (Exception)

1 scenario (1 failed)
6 steps (5 passed, 1 failed)
0m3.17s (11.15Mb)
```

That's as far as I got so far 😊

**Comment by mrsdizzie** [comment]
Thanks for checking! 

I had a moment to look into this and I think the problem is that sed writes in text mode by default on MacOS, so while adding `-i.bak` works, it can't actually add a BOM at the beginning of the file in text mode. I tried a few ways but couldn't actually figure out how to make sed write in binary mode and be able to use a regex like this on MacOS.

I gave up and tried awk which works and should be as portable:

`awk 'BEGIN {print "\xef\xbb\xbf"} {print}' wp-config.php > wp-config.php`

I tested that on the latest MacOS and a Debian 10.13 server and it works. I can send a PR if that change seems OK.
